docs/dev/how-to/estimate-gas.md

```diff
@@ -0,0 +1,399 @@
+
+1. Import the zkSync Era library or contract containing the required functionality. 
+
+2. Get a `Contract` object that represents the [`Messenger`](../developer-guides/system-contracts.md#l1messenger) contract.
+
+3. Transform the request into a raw bytes array.
+
+4. Use the [`sendToL1`](https://github.com/matter-labs/v2-testnet-contracts/blob/b8449bf9c819098cc8bfee0549ff5094456be51d/l2/system-contracts/interfaces/IL1Messenger.sol#L5) function from the `IL1Messenger.sol` interface, passing the message as a raw bytes array.
+
+```solidity
+function sendToL1(bytes memory _message) external returns (bytes32);
+```
+
+:::warning What is a message?
+- A message functions like an event on Ethereum. The difference is that a message publishes data on L1. 
+- Its Solidity representation:
+    ```solidity
+    struct L2Message {
+            address sender;
+            bytes data;
+            uint256 txNumberInblock;
+    }
+    ```
+:::
+
+5. The [`proveL2MessageInclusion`](https://github.com/matter-labs/v2-testnet-contracts/blob/b8449bf9c819098cc8bfee0549ff5094456be51d/l1/contracts/zksync/facets/Mailbox.sol#L46) function returns a boolean parameter indicating whether the message was sent successfully to L1.
+
+```solidity
+function proveL2MessageInclusion(
+    uint256 _blockNumber,
+    uint256 _index,
+    L2Message memory _message,
+    bytes32[] calldata _proof
+
+) public view returns (bool) {
+    return _proveL2LogInclusion(_blockNumber, _index, _L2MessageToLog(_message), _proof);   
+}
+```
+
+### Example
+
+::: code-tabs
+@tab Solidity
+```solidity
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+// Importing zkSync contract interface
+import "@matterlabs/zksync-contracts/l1/contracts/zksync/interfaces/IZkSync.sol";
+
+contract Example {
+
+    function sendMessageToL1() external returns(bytes32 messageHash) {
+        // Construct the message directly on the contract
+        bytes memory message = abi.encode(address(this));
+
+        messageHash = L1_MESSENGER_CONTRACT.sendToL1(message);
+    }
+
+  // NOTE: The zkSync contract implements only the functionality for proving that a message belongs to a block
+  // but does not guarantee that such a proof was used only once. That's why a contract that uses L2 -> L1
+  // communication must take care of the double handling of the message.
+  /// @dev mapping L2 block number => message number => flag
+  /// @dev Used to indicated that zkSync L2 -> L1 message was already processed
+  mapping(uint256 => mapping(uint256 => bool)) isL2ToL1MessageProcessed;
+
+  function consumeMessageFromL2(
+    // The address of the zkSync smart contract.
+    // It is not recommended to hardcode it during the alpha testnet as regenesis may happen.
```

Comment:
regenesis is unlikely, but yeah, it is better not to hardcode it

Reply:
we are out of alpha testnet, so I will remove this line